src/platform/K32W/KeyValueStoreManagerImpl.cpp

```diff
@@ -24,76 +24,189 @@
 #include <platform/internal/CHIPDeviceLayerInternal.h>
 
 #include <lib/support/CHIPMem.h>
+#include <list>
 #include <platform/K32W/K32WConfig.h>
 #include <platform/KeyValueStoreManager.h>
 #include <string>
 
 #include "PDM.h"
 
+#include <unordered_map>
+
 namespace chip {
 namespace DeviceLayer {
 namespace PersistedStorage {
 
 /* TODO: adjust this value */
-#define MAX_NO_OF_KEYS 255
+#define MAX_NO_OF_KEYS 20
 
 KeyValueStoreManagerImpl KeyValueStoreManagerImpl::sInstance;
 
-CHIP_ERROR KeyValueStoreManagerImpl::_Get(const char * key, void * value, size_t value_size, size_t * read_bytes_size,
-                                          size_t offset_bytes)
+/* hashmap having:
+ * 	- the matter key string as key;
+ * 	- internal PDM identifier as value;
+ */
+std::unordered_map<std::string, uint8_t> g_kvs_map;
+
+/* list containing used PDM identifiers */
```

Comment:
should this be a set?

Reply:
you could even do std::bitset and use only 32 bytes of memory.

should static assert that maximum number of keys fits in uint8_t since this is what you use for your set.

Reply:
Replaced list with set and added assert to check that the number of keys is at most kMaxNumberOfKeys.